client/app/src/main/java/com/example/criengine/Adapters/SearchBooksListAdapter.java

```diff
@@ -74,7 +77,13 @@ public void onSuccess(Profile profile) {
             @Override
             public void onClick(View v) {
                 Intent intent = new Intent(v.getContext(), NonOwnerBookViewActivity.class);
-                intent.putExtra("Page", RootActivity.PAGE.SEARCH);
+                // Book was clicked in search fragment
+                if (isSearchFragment) {
+                    intent.putExtra("Page", RootActivity.PAGE.SEARCH);
+                // Book was clicked in another activity
+                } else {
+                    intent.putExtra("Page", RootActivity.PAGE.OTHER);
```

Comment:
This will cause an error. We should use a page that's already there?
The PageViewer in RootActivity will look for a page of index 5

Reply:
This is for when we need to use super.onBackPressed() so we can go back to the profile the book came from

Reply:
https://github.com/CMPUT301F20T05/CriEngine/blob/a89efb386e54f9ca6c6e9fc47b4b6e52a524874e/client/app/src/main/java/com/example/criengine/Activities/RootActivity.java#L94